library/alloc/src/string.rs

```diff
@@ -1849,6 +1849,34 @@ impl String {
         let slice = self.vec.into_boxed_slice();
         unsafe { from_boxed_utf8_unchecked(slice) }
     }
+
+    /// Consumes and leaks the `String`, returning a mutable reference to the contents,
+    /// `&'a mut str`.
+    ///
+    /// This function is mainly useful for data that lives for the remainder of
+    /// the program's life. Dropping the returned reference will cause a memory
+    /// leak.
+    ///
+    /// # Examples
+    ///
+    /// Simple usage:
+    ///
+    /// ```
+    /// #![feature(string_leak)]
+    ///
+    /// pub fn main() {
+    ///     let x = String::from("bucket");
+    ///     let static_ref: &'static mut str = x.leak();
+    ///     assert_eq!(static_ref, "bucket");
+    /// }
+    /// ```
+    #[cfg(not(no_global_oom_handling))]
+    #[unstable(feature = "string_leak", issue = "102929")]
+    #[inline]
+    pub fn leak<'a>(self) -> &'a mut str {
```

Comment:
Is there any reason to use a generic `<'a>` instead of just `'static` here? I believe the only reason we use `<'a>` for `Vec<T>` is for situations where `T: !'static`.

```suggestion
    pub fn leak(self) -> &'static mut str {
```

Reply:
Actually, I think there is another reason. `Vec` is generic over `A: Allocator` and this may apply to `String` in the future. At that point, the generic lifetime would be bounded by that of the allocator. I don't know if adding the generic lifetime later would be a breaking change. Thoughts?

![image](https://user-images.githubusercontent.com/20015102/196505797-cf1f3e59-c27c-4c35-b4c7-2075f29cc12a.png)
![image](https://user-images.githubusercontent.com/20015102/196506260-da1ba1b9-605b-4f38-b33c-ffa24ece9041.png)